invertedai/large/common.py

```diff
@@ -0,0 +1,74 @@
+from typing import List, Optional, Tuple
+from pydantic import BaseModel
+
+from invertedai.common import AgentAttributes, AgentState, RecurrentState, Point
+
+class Region(BaseModel):
+    """
+    A region in a map used to divide a large simulation into smaller parts.
+
+    See Also
+    --------
+    AgentAttributes
+    """
+
+    center: Optional[Point] = None #: The center of the region if such a concept is relevant (e.g. center of a square, center of a rectangle)
+    size: Optional[float] = None #: Side length of the region for the default interpretation of a region as a square
+    agent_states: Optional[List[AgentState]] = [] #: A list of existing agents within the region
+    agent_attributes: Optional[List[AgentAttributes]] = [] #: The attributes of agents that exist within the region or that will be initialized within the region
+    recurrent_states: Optional[List[RecurrentState]] = [] #: Recurrent states of the agents contained within the region
+
+    @classmethod
+    def init_square_region(
+        cls, 
+        center, 
+        size = 100, 
+        agent_states = [], 
+        agent_attributes = [], 
+        recurrent_states = []
+    ):
+        for agent in agent_states:
+            assert cls.check_point_in_bounding_box(cls,agent.center), f"Existing agent states at position {agent.center} must be located within the region."
+
+        return cls(
+            center=center, 
+            size=size,
+            agent_states=agent_states, 
+            agent_attributes=agent_attributes, 
+            recurrent_states=recurrent_states 
+        )
+
```

Comment:
A `copy` method would be a good idea for future proofing.

Reply:
Done.